### PR TITLE
Fix: Deployment Account Step Function Time outs

### PR DESCRIPTION
**Pull request by StewartW**
*Issue #, if available:* #400 

*Description of changes:*
Refactored state machine slightly to move account iteration logic into
a map, lambda function then does region checks for each account as before.

No change to function logic bar removing the for account_id in account_ids
loop.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/enable_cross_account_access.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def update_iam(role, s3_bucket, kms_key_arn, role_policies):
 
 
 def lambda_handler(event, _):
+    LOGGER.debug(event)
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/enable_cross_account_access.py:39]: Can you remove this debug line, or only expose the data that really needs to be present + explaining where we are at / what we are processing?

Like
```bash
"Setting up cross-account access for ${account_id} in ${regions}"
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 2]
```diff
@@ -1118,12 +1118,29 @@ Resources:
...
                     "Default": "EnableCrossAccountAccess"
                 },
-                "EnableCrossAccountAccess": {
-                    "Type": "Task",
-                    "Resource": "${EnableCrossAccountAccess.Arn}",
-                    "Next": "UpdateDeploymentPipelines",
-                    "TimeoutSeconds": 900
-                },
+                "EnableCrossAccountAccessMap": {
+                      "Type": "Map",
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:1122]: The indentation is off, could you fix that?
Probably happened when the partition PR got merged.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 3]
```diff
@@ -1118,12 +1118,29 @@ Resources:
...
+                      },
+                      "ItemsPath": "$.account_ids",
+                      "Parameters": {
+                        "deployment_account_region.$": "$.deployment_account_region",
+                        "deployment_account_id.$": "$.deployment_account_id",
+                        "account_id.$": "$$.Map.Item.Value",
+                        "regions.$": "$.regions"
+                      },
+                      "ResultPath": null
+                    }
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:1143]: I think there is missing comma here:
```suggestion
                    },
```

Comment (sbkok) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:1143]: Oddly enough, the next line indents on a slightly different line it seems. Can you check this?

**Review by sbkok — CHANGES_REQUESTED** [review]
Thanks for working on this issue.
Could you look at the attached comments?

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/enable_cross_account_access.py (hunk 1 above)
Comment (StewartW) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/enable_cross_account_access.py:39]: I'll just get rid of it. It's a hang over from debugging the step function map.

**Review by sbkok — APPROVED** [review]
LGTM, thank you!

**Review by dsudduth — APPROVED** [review]
The latest round of changes look great! Thank you for the fix!

**Pull request merged**